modin/test/test_partition_api.py

```diff
@@ -54,27 +54,42 @@
 
 
 @pytest.mark.parametrize("axis", [None, 0, 1])
-def test_unwrap_partitions(axis):
+@pytest.mark.parametrize("reverse_index", [True, False])
+@pytest.mark.parametrize("reverse_columns", [True, False])
+def test_unwrap_partitions(axis, reverse_index, reverse_columns):
     data = test_data["int_data"]
-    df = pd.DataFrame(data)
 
+    def get_df(lib, data):
+        df = lib.DataFrame(data)
+        if reverse_index:
+            df.index = df.index[::-1]
+        if reverse_columns:
+            df.columns = df.columns[::-1]
+        return df
+
+    df = get_df(pd, data)
+    # `df` should not have propagated the index and column updates to its
+    # partitions yet. The partitions of `expected_df` should have the updated
+    # metadata because we construct `expected_df` directly from the updated
+    # pandas dataframe.
+    expected_df = pd.DataFrame(get_df(pandas, data))
+    expected_partitions = expected_df._query_compiler._modin_frame._partitions
     if axis is None:
-        expected_partitions = df._query_compiler._modin_frame._partitions
         actual_partitions = np.array(unwrap_partitions(df, axis=axis))
         assert (
```

Comment:
Couldn't this assert just check that shape == shape rather than checking each axis of shape individually?

Reply:
Sure, I can fix that while I'm here. Done.